### PR TITLE
copy for homepage

### PR DESCRIPTION
**Pull request by depombo**

### CODE REVIEW

File: src/homepage.md [hunk 1]
```diff
@@ -61,34 +61,38 @@ on app.start {
...
+    const [authUsers, dbUsers, crmUsers] = await Promise.all([
+      Auth.getAll(),
+      Store.getAllUsers(),
+      Crm.getAllUsers()
+    ]);
+    const authUids = authUsers.map(u => u['id']);
+    const dbUids = dbUsers.map(u => u['uid']);
+    const crmUids = crmUsers.map(u => u['uid']);
+    const validUids = authUids.filter(v => dbUids.includes(v) && crmUids.includes(v))
+  }
```
Comment (dfellis) [inline src/homepage.md]: These should both return the `validUids` variable at the end.

File: src/homepage.md [hunk 2]
```diff
@@ -61,34 +61,38 @@ on app.start {
...
   /* ALAN automatically executes IO in parallel when possible */
-  const authUids = Auth.getAllUsers().map(fn (u: AuthUser) = u.id)
-  const dbUsers = Store.getAllUsers().map(fn (u: User) = u.uid)
-  const crmUsers = Crm.getAllUsers().map(fn (u: CrmUser) = u.uid)
-  const validUids = authUids.filter(v => dbUids.has(v) && crmUids.has(v))
+  fn getValidUids() {
+    const authUids = Auth.getAllUsers().map(fn (u: AuthUser) = u.id)
+    const dbUsers = Store.getAllUsers().map(fn (u: User) = u.uid)
+    const crmUsers = Crm.getAllUsers().map(fn (u: CrmUser) = u.uid)
+    const validUids = authUids.filter(v => dbUids.has(v) && crmUids.has(v))
```
Comment (dfellis) [inline src/homepage.md:68]: Oopsie: `=>` should be `=`

Comment (dfellis) [reply src/homepage.md:68]: (and should have the prefix `fn` and `int` type for `v`

File: src/homepage.md [hunk 3]
```diff
@@ -61,34 +61,38 @@ on app.start {
       <li class="carousel__slide">
         <pre class="code-border"><code class="language-javascript">
   /* ALAN automatically executes IO in parallel when possible */
-  const authUids = Auth.getAllUsers().map(fn (u: AuthUser) = u.id)
-  const dbUsers = Store.getAllUsers().map(fn (u: User) = u.uid)
-  const crmUsers = Crm.getAllUsers().map(fn (u: CrmUser) = u.uid)
-  const validUids = authUids.filter(v => dbUids.has(v) && crmUids.has(v))
+  fn getValidUids() {
+    const authUids = Auth.getAllUsers().map(fn (u: AuthUser) = u.id)
```
Comment (dfellis) [inline src/homepage.md:65]: Technically all closure functions passed to array methods (or any opcode, really) require the return type to be explicitly defined. I don't know if we'll get that restriction lifted before the website is public, so we probably want to add those annotations here.

File: src/homepage.md (hunk 2 above)
Comment (depombo) [reply src/homepage.md:68]: good catch, thanks!

File: src/homepage.md (hunk 1 above)
Comment (depombo) [reply src/homepage.md]: doh

File: src/homepage.md [hunk 4]
```diff
@@ -62,10 +62,10 @@ on app.start {
...
   /* ALAN automatically executes IO in parallel when possible */
   fn getValidUids() {
-    const authUids = Auth.getAllUsers().map(fn (u: AuthUser) = u.id)
-    const dbUsers = Store.getAllUsers().map(fn (u: User) = u.uid)
-    const crmUsers = Crm.getAllUsers().map(fn (u: CrmUser) = u.uid)
-    const validUids = authUids.filter(v => dbUids.has(v) && crmUids.has(v))
+    const authUids = Auth.getAllUsers().map(fn (u: AuthUser): int = u.id)
+    const dbUsers = Store.getAllUsers().map(fn (u: User): int = u.uid)
+    const crmUsers = Crm.getAllUsers().map(fn (u: CrmUser): int = u.uid)
+    return authUids.filter(fn (v: int): int = dbUids.has(v) && crmUids.has(v))
```
Comment (dfellis) [inline src/homepage.md:68]: Should be `fn (v: int) : bool = ...`

File: src/homepage.md [hunk 5]
```diff
@@ -35,71 +39,135 @@ on app.start {
...
+    const crmUids = crmUsers.map(u => u['uid']);
+    return authUids.filter(v => dbUids.includes(v) && crmUids.includes(v))
+  }
+        </code></pre>
+      </li>
+      <li class="carousel__slide">
+        <pre class="code-border"><code class="language-golang">
+  /* ALAN automatically executes CPU operations in parallel when sensible */
+  fn sumMaybeConcurrent(nums: Array&lt;int&gt;): int {
+    return nums.reduce(fn (accum: int, val: int): int = accum + val)
```
Comment (dfellis) [inline src/homepage.md:90]: Not blocking: I can't wait until we fix the issue preventing us from just writing `fn sumMaybeConcurrent(nums: Array<int>) = nums.reduce(add)`

Comment (depombo) [reply src/homepage.md:90]: 💯

**Pull request merged**